### PR TITLE
Support for `#` prefix

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Add support for marking annotations with `#` character. It helps to properly track the annotation beginning for annotation types of `invariant` or `define` (these can be used as a regular words). This also would simplify life for the tool developers as annotations would have begin (`#`) and end (`;`) markers.
- [x] Added deprecation notice for annotations without `#` prefix.
- [x] Added relevant test file and tweaked existing tests to not fail due to a deprecation notice.

Regards.

### CODE REVIEW

File: src/instrumenter/annotations.ts [hunk 1]
```diff
@@ -395,7 +395,7 @@ class AnnotationExtractor {
 
         const result: AnnotationMetaData[] = [];
 
-        const rx = /\s*(\*|\/\/\/)\s*(if_succeeds|if_updated|if_assigned|invariant|define\s*[a-zA-Z0-9_]*\s*\([^)]*\))/g;
+        const rx = /\s*(\*|\/\/\/)(\s*#)?\s*(if_succeeds|if_updated|if_assigned|invariant|define\s*[a-zA-Z0-9_]*\s*\([^)]*\))/g;
```
Comment (cd1m0) [inline src/instrumenter/annotations.ts:398]: I don't think we should allow space between the `#` and the keyword. (So no `#  if_succeeds`)

File: src/spec-lang/expr_grammar.pegjs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Top-level rules
 Annotation
-    = StartingWhiteSpace annotation: (Invariant
+    = StartingWhiteSpace ('#' __)? annotation: (Invariant
```
Comment (cd1m0) [inline src/spec-lang/expr_grammar.pegjs:3]: Same here - remove the '__' after '#'

**Review by cd1m0 — CHANGES_REQUESTED** [review]
Good work! Just two things:

1. Fix the space thing after #

2. Lets tweak this PR so that when we encounter a correctly parsed annotation without the `#` we emit a warning to the user that "Annotations without `#` will be eventually deprecated. Please add a `#` to your annotations"

File: src/spec-lang/expr_grammar.pegjs (hunk 2 above)
Comment (blitz-1306) [reply src/spec-lang/expr_grammar.pegjs]: Addressed in https://github.com/ConsenSys/scribble/pull/44/commits/02b98f0d598cab8ac5fc53764fc058c3f88404a8

File: src/instrumenter/annotations.ts (hunk 1 above)
Comment (blitz-1306) [reply src/instrumenter/annotations.ts]: Addressed in https://github.com/ConsenSys/scribble/pull/44/commits/02b98f0d598cab8ac5fc53764fc058c3f88404a8

**Review by cd1m0 — APPROVED** [review]
LGTM

**Pull request merged**